### PR TITLE
Improve the Config context

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

- make the config object `ReadOnly` so it’s not inadvertently mutated
- reduce serialised size in the DOM by using a single JSON object
- add the `editionId` to the config
- use the pattern for [`LatestLinks`](https://github.com/guardian/dotcom-rendering/pull/11742/files#diff-8c7596d837d0faed626e58361475851f5c3a50c0f86fe073dc7d489d1c6b0e48) and squash potential bugs

## Why?

This pattern, introduced in #8704, has been very useful. This addressed initial concerns from @georgeblahblah, @ioanna0 and myself about redundant data in the HTML. Adding the edition can help reduce deeply drilled props, and reduce bugs, such as in `CardAge`

### CODE REVIEW

File: dotcom-rendering/src/client/islands/getConfig.ts [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
+
+	const serialised = document.querySelector('script#config')?.innerHTML;
 
 	try {
 		if (!serialised) {
-			throw Error('Unable to fetch config attribute from marker element');
+			throw Error('Unable to fetch config attribute from #config');
 		} else {
-			return JSON.parse(serialised) as Config;
+			config = JSON.parse(serialised) as Config;
```
Comment (mxdvl) [inline dotcom-rendering/src/client/islands/getConfig.ts:19]: For the keen beans amongs you: consider using valibot to parse this?

File: dotcom-rendering/src/server/htmlPageTemplate.ts [hunk 2]
```diff
@@ -75,6 +77,11 @@ export const htmlPageTemplate = (props: WebProps | AppProps): string => {
 	 */
 	const windowGuardian = escapeData(JSON.stringify(guardian));
 
+	if (config.renderingTarget === 'Apps') {
+		config.darkModeAvailable;
+		config.updateLogoAdPartnerSwitch;
+	}
```
Comment (arelra) [inline dotcom-rendering/src/server/htmlPageTemplate.ts:83]: Was the intention to assign values here?

Comment (mxdvl) [reply dotcom-rendering/src/server/htmlPageTemplate.ts:83]: haha no just testing values in the IDE!

File: dotcom-rendering/src/types/configContext.ts [hunk 3]
```diff
@@ -8,15 +9,17 @@ import type { RenderingTarget } from './renderingTarget';
...
 export type Config =
-	| {
+	| Readonly<{
 			renderingTarget: Extract<RenderingTarget, 'Web'>;
 			darkModeAvailable: boolean;
 			updateLogoAdPartnerSwitch: boolean;
 			assetOrigin: AssetOrigin;
-	  }
-	| {
+			editionId: EditionId;
```
Comment (abeddow91) [inline dotcom-rendering/src/types/configContext.ts:17]: We have a rule in the sand for our config use that [Context should be considered **global, static, and immutable** ](https://github.com/guardian/dotcom-rendering/blob/85e286cc5b36697f49e9373b993b3ad3a598a641/dotcom-rendering/docs/architecture/028-react-context-api.md?plain=1#L15C5-L15C68) - Does this violate that?

**Pull request merged**